src/Features/CSharp/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.cs

```diff
@@ -536,14 +536,17 @@ private static SyntaxNode GetTopMostParentingExpression(ExpressionSyntax express
                             newInitializerSymbolInfo = newSemanticModelForInlinedDocument.GetSymbolInfo(inlinedNode, cancellationToken);
                             if (!SpeculationAnalyzer.SymbolInfosAreCompatible(originalInitializerSymbolInfo, newInitializerSymbolInfo, performEquivalenceCheck: true))
                             {
-                                if (replacementNodesWithChangedSemantics == null)
-                                {
-                                    replacementNodesWithChangedSemantics = new Dictionary<SyntaxNode, SyntaxNode>();
-                                }
-
+                                replacementNodesWithChangedSemantics = replacementNodesWithChangedSemantics ?? new Dictionary<SyntaxNode, SyntaxNode>();
                                 replacementNodesWithChangedSemantics.Add(inlinedNode, originalNode);
                             }
                         }
+
+                        // Verification: Do not inline a variable into the left side of a deconstruction-assignment
+                        if (IsInDeconstructionAssignmentLeft(innerInitializerInInlineNode))
+                        {
+                            replacementNodesWithChangedSemantics = replacementNodesWithChangedSemantics ?? new Dictionary<SyntaxNode, SyntaxNode>();
```

Comment:
j/k :)

Reply:
That's mean :-)

---
In reply to: [138161999](https://github.com/dotnet/roslyn/pull/21890#discussion_r138161999) [](ancestors = 138161999)